bpo-36013: Fix the interactive shell SIGINT test.

gpshead opened the pull request:
First attempt: switch shell interupt test to zsh.

if this doesn't work out, i'll just delete this test.  it's a nice bonus integration test to avoid a regression of the actual interactive mode ^C'd process detection rather than the weak test that looks only at exit status (not what shells do).

<!-- issue-number: [bpo-36013](https://bugs.python.org/issue36013) -->
https://bugs.python.org/issue36013
<!-- /issue-number -->

gpshead committed:
bpo-36013: Fix the interactive shell SIGINT test.

First attempt: switch shell interupt test to zsh.

pablogsal commented on Lib/test/test_signal.py:
```diff
@@ -90,35 +90,40 @@ def test_keyboard_interrupt_exit_code(self):
-            ["bash", "-ic",
+            ["zsh", "-ic",
+             "echo TEST  START 1>&2; "
              f"{sys.executable} -c 'import os,signal; os.kill(os.getpid(), signal.SIGINT)'; "
```
From the issue:

> Could it be a race condition the signal delivery?
>Technically, the process sends itself a signal and never waits for it, it just exits. The signal can arrive after the process has exited normally. Or I am missing something obvious in this reasoning?

gpshead replied:
it is possible (which is why i was thinking maybe flaky) but I _suspect_ more is going on here as it only seems unreliable on CI and buildbot systems, and seems reliable from our login shells.  I'm opting to not try and figure it out.  testing interactive things within in automation isn't worth the effort for this issue. :)

gpshead replied:
the other test above this one also uses the same self-killing code fwiw.

gpshead committed:
delete the interactive shell behavior test.

gpshead committed:
make the existing more readable and robust.

Unclear if this is necessary, but if there is a race condition between
delivering ourselves a SIGINT and the CPython SIGINT handler that raises
KeyboardInterrupt happening before our process could exit, this should
avoid it.  I made it easier to read and maintain regardless.

eamanu reviewed (approved):
lgtm
